frontend/controller/service-worker.js

```diff
@@ -98,6 +98,10 @@ sbp('sbp/selectors/register', {
       ])
       const swRegistration = await navigator.serviceWorker.register(`/assets/js/sw-primary.js?${params}`, { scope: '/' })
 
+      if (swRegistration.active == null && swRegistration.installing == null && swRegistration.waiting == null) {
+        throw new Error('No valid service worker found')
```

Comment:
This probably happens if the SW immediately crashed after installation (e.g., trying to access `indexedDB` if it's undefined), or that's my theory. In any case, we can't proceed when this happens and `.update()` below will fail. The goal of this check is to provide a more useful error message.